python/pyarrow/type_traits.pxi

```diff
@@ -0,0 +1,721 @@
+    Useful for ensuring that a given data type is a floating type before performing
+    operations that are only valid for floating point numbers.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+
+    Returns
+    -------
+    bool
+        True if `data_type` is a floating type, False otherwise.
+    """
+    return is_floating(data_type.id)
+
+
+def is_numeric_type(data_type):
+    """
+    Check if the data type is a numeric type.
+
+    This function checks whether the `data_type` is a numeric type, which includes
+    integers and floating point numbers with specific bit widths. Integer types
+    include signed and unsigned integers of various bit widths (8, 16, 32, 64 bits),
+    while floating point types include floating point numbers of various bit widths
+    (16, 32, 64 bits).
+
+    Useful for ensuring that a given data type is a numeric type before performing
+    operations that are only valid for numeric types.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+
+    Returns
+    -------
+    bool
+        True if `data_type` is a numeric type, False otherwise.
+    """
+    return is_numeric(data_type.id)
+
+
+def is_decimal_type(data_type):
+    """
+    Check if the data type is a decimal type.
+
+    This function checks whether the `data_type` is a decimal type, which includes
+    fixed-point decimal numbers with specific precision and scale.
+
+    Useful for ensuring that a given data type is a decimal type before performing
+    operations that are only valid for decimal numbers.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+
+    Returns
+    -------
+    bool
+        True if `data_type` is a decimal type, False otherwise.
+    """
+    return is_decimal(data_type.id)
+
+
+def is_run_end_type_py(data_type):
+    """
+    Check if the data type is a run end type.
+
+    This function checks whether the `data_type` is a run end type, which includes
+    integers of various bit widths (16, 32, 64 bits).
+
+    Useful for ensuring that a given data type is a run end type before performing
+    operations that are only valid for run end types.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+
+    Returns
+    -------
+    bool
+        True if `data_type` is a run end type, False otherwise.
+    """
+    return is_run_end_type(data_type.id)
+
+
+def is_primitive_type(data_type):
+    """
+    Check if the data type is a primitive type.
+
+    This function checks whether the `data_type` is a primitive type, which includes
+    integers, floating point numbers, dates (days since the UNIX epoch and milliseconds
+    since the UNIX epoch), times (seconds and milliseconds since midnight),
+    timestamp (milliseconds since the UNIX epoch), and duration (elapsed time in
+    seconds, milliseconds, microseconds, and nanoseconds), and intervals (months,
```

Comment:
```suggestion
    timestamp (milliseconds since the UNIX epoch), duration (elapsed time in
    seconds, milliseconds, microseconds, and nanoseconds) and intervals (months,
```

Booleana are also included here.

Also, I think it would be enough to only keep the content without the details in `(...)`.

Reply:
I updated docstring as you mentioned